src/geobr.jl

```diff
@@ -0,0 +1,537 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+Provides functions to download data from the GeoBR database:
+
+`GeoBR.state`
+`GeoBR.municipality`
+`GeoBR.region`
+`GeoBR.country`
+`GeoBR.amazon`
+`GeoBR.biomes`
+`GeoBR.disasterriskarea`
+`GeoBR.healthfacilities`
+`GeoBR.indigenousland`
+`GeoBR.metroarea`
+`GeoBR.neighborhood`
+`GeoBR.urbanarea`
+`GeoBR.weightingarea`
+`GeoBR.mesoregion`
+`GeoBR.microregion`
+`GeoBR.intermediateregion`
+`GeoBR.immediateregion`
+`GeoBR.municipalseat`
+`GeoBR.censustract`
+`GeoBR.statisticalgrid`
+`GeoBR.conservationunits`
+`GeoBR.semiarid`
+`GeoBR.schools`
+`GeoBR.comparableareas`
+`GeoBR.urbanconcentrations`
+`GeoBR.poparrangements`
+`GeoBR.healthregion`
+"""
+module GeoBR
+
+using GeoIO
+using DataDeps
+using CSV
+using Tables
+using TableTransforms
+
+const APIVERSIONS = (v"1.7.0",)
+
+"""
+    download(url; version=last(APIVERSIONS))
+
+(Down)load data for the specified `url` and `version` of the dataset.
+"""
+function download(url; version=last(APIVERSIONS))
+  ID = "GeoBR_$(version)_$(basename(url))"
+  try
+    # if data is already on disk
+    # we just return the path
+    @datadep_str ID
+  catch
+    # otherwise we register the data
+    # and download using DataDeps.jl
+    register(DataDep(
+      ID,
+      """
+      Data for GeoBR project.
+      Source: $url
+      """,
+      url,
+      Any
+    ))
+    @datadep_str ID
+  end
+  joinpath(ID, basename(url))
+end
+
+"""
+    downloadmeta(version=last(APIVERSIONS))
+
+(Down)load metadata for the specified `version` of the dataset.
+"""
+downloadmeta(version=last(APIVERSIONS)) = download("http://www.ipea.gov.br/geobr/metadata/metadata_$(version)_gpkg.csv"; version)
+
+"""
+    geturl(csv, entity, year, code)
+
+Retrieve rows of `csv` for the given `entity`, `year` and `code`.
+"""
+function geturl(csv, entity, year, code)
+  # TODO
+end
+
+"""
+    get(entity, year; code=nothing, version=last(APIVERSIONS), kwargs...)
+
+Load geographic data for given `entity` and `year`. Optionally specify
+`code` or abbreviation and dataset `version`. The `kwargs` are forwarded
+to `GeoIO.load`.
+"""
+function get(entity, year; code=nothing, version=last(APIVERSIONS), kwargs...)
+  csv = CSV.File(downloadmeta(version))
+  url = geturl(csv, entity, year, code)
+  GeoIO.load(download(url; version); kwargs...)
```

Comment:
This is the general logic. The geturl function should encapsulate all the tricks to get the url from given parameters.